modules/platforms/dotnet/Apache.Ignite/Internal/Table/Serialization/ObjectSerializerHandler.cs

```
@@ -0,0 +1,149 @@
+    internal class ObjectSerializerHandler<T> : IRecordSerializerHandler<T>
+        where T : class
+    {
+        /// <inheritdoc/>
+        public T Read(ref MessagePackReader reader, Schema schema, bool keyOnly = false)
+        {
+            // TODO: Emit code for efficient serialization (IGNITE-16341).
+            var columns = schema.Columns;
+            var count = keyOnly ? schema.KeyColumnCount : columns.Count;
+            var res = Activator.CreateInstance<T>();
+            var type = typeof(T);
+
+            for (var index = 0; index < count; index++)
+            {
+                if (reader.TryReadNoValue())
+                {
+                    continue;
+                }
+
+                var col = columns[index];
+                var prop = GetPropertyIgnoreCase(type, col.Name);
+
+                if (prop != null)
+                {
+                    var value = reader.ReadObject(col.Type);
+                    prop.SetValue(res, value);
+                }
+                else
+                {
+                    reader.Skip();
+                }
+            }
+
+            return (T)(object)res;
+        }
+
+        /// <inheritdoc/>
+        public T ReadValuePart(PooledBuffer buf, Schema schema, T key)
+        {
+            // TODO: Emit code for efficient serialization (IGNITE-16341).
+            // Skip schema version.
+            var r = buf.GetReader();
+            r.Skip();
+
+            var columns = schema.Columns;
+            var res = Activator.CreateInstance<T>();
+            var type = typeof(T);
+
+            for (var i = 0; i < columns.Count; i++)
+            {
+                var col = columns[i];
+                var prop = GetPropertyIgnoreCase(type, col.Name);
+
+                if (i < schema.KeyColumnCount)
+                {
+                    if (prop != null)
+                    {
+                        prop.SetValue(res, prop.GetValue(key));
+                    }
```

Here we just silently ignore a case when user class does not have a column that present in schema. Is this behavior discussed somewhere?

---

Same behavior as in Java. I could not find any discussions on this topic, but it seems reasonable to me.

---

Ok, but maybe we still need to discuss that.